core/src/test/java/org/apache/calcite/test/enumerable/EnumerableUncollectTest.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.test.enumerable;
+
+import org.apache.calcite.config.CalciteConnectionProperty;
+import org.apache.calcite.config.Lex;
+import org.apache.calcite.test.CalciteAssert;
+
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+class EnumerableUncollectTest {
+
+  @Test void simpleUnnestArray() {
+    final String sql = "select * from UNNEST(array[3, 4]) as T2(y)";
+    tester()
+        .query(sql)
+        .returnsUnordered(
```

Comment:
Should these all really be unordered? I'm not sure what other implementations do, but I would have expected array elements to remain in order (same for the tests below).

Reply:
If I am not mistaken, our UNNEST implementation keeps the order, however SQL standard says that UNNEST does not guarantee to keep order, that is why the "WITH ORDINALITY" exists. This the reason why `returnsUnordered` is used here.